treeherder/webapp/graphql/helpers.py

```diff
@@ -0,0 +1,66 @@
+from graphene_django.filter import DjangoFilterConnectionField
+
+
+def collect_fields(node):
+    """Get all the unique field names that are eligible for optimization"""
+    fields = set()
+    for leaf in node:
+        if leaf.get('kind', None) == "Field":
+            fields.add(leaf["name"]["value"])
+
+        if leaf.get("selection_set", None):
+            fields = fields.union(collect_fields(leaf["selection_set"]["selections"]))
+
+    return fields
+
+
+def optimize(qs, info_dict, field_map):
```

Comment:
This and `collect_fields` seem like things that should be built into graphene-django -- would you mind filing an upstream ticket and linking to it here?